source/common/tracing/zipkin/tracer.h

```diff
@@ -78,18 +86,40 @@ class Tracer : public TracerInterface {
   void reportSpan(Span&& span) override;
 
   /**
+   * @return the service-name attribute associated with the Tracer.
+   */
+  const std::string& serviceName() const { return service_name_; }
+
+  /**
+   * @return the pointer to the address object associated with the Tracer.
+   */
+  const Network::Address::InstanceConstSharedPtr address() const { return address_; }
+
+  /**
    * Associates a Reporter object with this Tracer.
    */
   void setReporter(ReporterPtr reporter);
 
   /**
+   * @return the reporter associated with the Tracer.
+   * This is only implemented to support the move constructor.
+   */
+  ReporterPtr reporter() { return std::move(reporter_); }
```

Comment:
If I do that the code does not pass asan/tsan checks. The following error is reported:
`moving a temporary object prevents copy elision [-Werror,-Wpessimizing-move]`